src/tuesday/core/summaries/powerspectra.py

```diff
@@ -13,58 +18,121 @@
 from scipy.interpolate import RegularGridInterpolator
 
 
-def calculate_ps(  # noqa: C901
-    lc,
-    lc_redshifts,
-    box_length,
-    box_side_shape=None,
-    zs=None,
-    chunk_size=None,
-    chunk_skip=37,
-    calc_2d=True,
-    nbins=50,
-    k_weights=ignore_zero_ki,
-    postprocess=True,
-    kpar_bins=None,
-    log_bins=True,
-    crop=None,
-    calc_1d=False,
-    nbins_1d=14,
-    calc_global=False,
-    mu=None,
-    bin_ave=True,
-    interp=None,
-    prefactor_fnc=power2delta,
-    interp_points_generator=None,
-    get_variance=False,
+@dataclass(frozen=True)
+class PowerSpectrum:
+    """Class to hold the power spectrum data."""
+
+    ps_1d: np.ndarray | None = None
+    ps_2d: np.ndarray | None = None
+    k: np.ndarray | None = None
+    kperp: np.ndarray | None = None
+    kpar: np.ndarray | None = None
+    redshifts: np.ndarray | None = None
+    Nmodes_1D: np.ndarray | None = None
+    Nmodes_2D: np.ndarray | None = None
+    var_1d: np.ndarray | None = None
+    var_2d: np.ndarray | None = None
+
+
+def get_chunk_indices(
+    lc_redshifts: np.ndarray,
+    box_side_shape: int,
```

Comment:
I don't think it's necessary to have this argument -- just let the user specify the chunk_size. Maybe maybe you could instead accept the box itself, and take the default for chunk_size to be the `box.shape[0]`.